test/framework/integration_framework/fake_peer/behaviour/behaviour.cpp

```diff
@@ -21,29 +21,56 @@ namespace integration_framework {
       // Hint: such calls would precede the derived class construction.
       fake_peer_wptr_ = fake_peer;
       log_ = std::move(log);
+      std::weak_ptr<Behaviour> weak_this = shared_from_this();
       // subscribe for all messages
       subscriptions_.emplace_back(
           getFakePeer().getMstStatesObservable().subscribe(
-              [this](const auto &message) {
-                this->processMstMessage(message);
+              [weak_this,
+               weak_fake_peer = fake_peer_wptr_](const auto &message) {
+                auto me_alive = weak_this.lock();
```

Comment:
Code of locking and checking is duplicated 5 times. I think better to refactor it.